In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd

# 파일 경로 지정 (.xlsx 파일)
file_path = '/content/drive/MyDrive/프로젝트/KCGS_평가등급_업데이트_완료.xlsx'

# 엑셀 파일 불러오기
df = pd.read_excel(file_path)

# 데이터 개요 확인
print("=== 데이터 정보 ===")
df.info()

print("\n=== 상위 5개 데이터 ===")
df.head()

=== 데이터 정보 ===
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 377 entries, 0 to 376
Data columns (total 20 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   기업명           377 non-null    object 
 1   기업코드          377 non-null    int64  
 2   평가년도          377 non-null    int64  
 3   ESG등급         377 non-null    object 
 4   환경            377 non-null    object 
 5   사회            377 non-null    object 
 6   지배구조          377 non-null    object 
 7   ESG 등급조정      0 non-null      float64
 8   환경 등급조정       0 non-null      float64
 9   사회 등급조정       0 non-null      float64
 10  지배구조 등급조정     1 non-null      object 
 11  시장구분          377 non-null    object 
 12  업종명           377 non-null    object 
 13  종가            101 non-null    float64
 14  대비            101 non-null    float64
 15  등락률           101 non-null    float64
 16  시가총액          376 non-null    float64
 17  지속가능경영보고서_발행  377 non-null    object 
 18  기업지배구조보고서_발행  3

,기업명,기업코드,평가년도,ESG등급,환경,사회,지배구조,ESG 등급조정,환경 등급조정,사회 등급조정,지배구조 등급조정,시장구분,업종명,종가,대비,등락률,시가총액,지속가능경영보고서_발행,기업지배구조보고서_발행,자산총액
0,에스앤에스텍,101490,2022,C,D,D,B,NaN,NaN,NaN,NaN,KOSDAQ,전기·전자,NaN,NaN,NaN,26350.0,N,N,2.267503e+11
1,에스앤에스텍,101490,2023,C,D,D,B+,NaN,NaN,NaN,NaN,KOSDAQ,전기·전자,NaN,NaN,NaN,45300.0,N,N,2.567587e+11
2,에스앤에스텍,101490,2024,C,D,D,B+,NaN,NaN,NaN,NaN,KOSDAQ,전기·전자,NaN,NaN,NaN,25600.0,N,N,2.970317e+11
3,에스앤에스텍,101490,2025,C,D,D,B+,NaN,NaN,NaN,NaN,KOSDAQ,전기·전자,48300.0,50.0,0.1,48300.0,N,N,3.930417e+11
4,일진전기,103590,2022,D,D,C,D,NaN,NaN,NaN,NaN,KOSPI,전기·전자,NaN,NaN,NaN,5020.0,N,N,8.395132e+11


In [ ]:
# 1. 2025년 및 '전기·전자' 업종 필터링
df_elec_2025 = df[(df['평가년도'] == 2025) & (df['업종명'] == '전기·전자')]

# 2. 모든 컬럼이 생략 없이 출력되도록 pandas 옵션 설정
pd.set_option('display.max_columns', None)

# 3. 전체 기업 수(행 개수) 출력
print(f"2025년 전기·전자 업종 전체 기업 수: {len(df_elec_2025)}개\n")

# 4. 전체 데이터 출력 (상위 데이터만 확인하시려면 df_elec_2025.head() 사용)
df_elec_2025

2025년 전기·전자 업종 전체 기업 수: 101개



,기업명,기업코드,평가년도,ESG등급,환경,사회,지배구조,ESG 등급조정,환경 등급조정,사회 등급조정,지배구조 등급조정,시장구분,업종명,종가,대비,등락률,시가총액,지속가능경영보고서_발행,기업지배구조보고서_발행,자산총액
3,에스앤에스텍,101490,2025,C,D,D,B+,NaN,NaN,NaN,NaN,KOSDAQ,전기·전자,48300.0,50.0,0.10,48300.0,N,N,3.930417e+11
7,일진전기,103590,2025,D,C,C,D,NaN,NaN,NaN,NaN,KOSPI,전기·전자,54400.0,-900.0,-1.63,54400.0,N,Y,1.525997e+12
11,LX세미콘,108320,2025,A,A,A+,B+,NaN,NaN,NaN,NaN,KOSPI,전기·전자,49150.0,-400.0,-0.81,49150.0,Y,Y,1.399843e+12
15,주성코퍼레이션,109070,2025,D,C,C,D,NaN,NaN,NaN,NaN,KOSPI,전기·전자,792.0,-10.0,-1.25,792.0,N,N,4.198561e+10
21,하나머티리얼즈,166090,2025,B,B+,A+,D,NaN,NaN,NaN,NaN,KOSDAQ,전기·전자,43100.0,-150.0,-0.35,43100.0,N,N,6.856062e+11
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
363,이엠텍,91120,2025,D,D,D,C,NaN,NaN,NaN,NaN,KOSDAQ,전기·전자,8090.0,160.0,2.02,8090.0,N,N,3.405336e+11
367,파트론,91700,2025,D,C,D,D,NaN,NaN,NaN,NaN,KOSDAQ,전기·전자,6970.0,20.0,0.29,6970.0,N,N,7.751037e+11
372,KEC,92220,2025,D,C,D,D,NaN,NaN,NaN,NaN,KOSPI,전기·전자,696.0,9.0,1.31,3480.0,N,N,4.489232e+11
373,칩스앤미디어,94360,2025,C,D,C,B,NaN,NaN,NaN,NaN,KOSDAQ,전기·전자,18050.0,-190.0,-1.04,18050.0,N,N,1.063037e+11


In [ ]:
import pandas as pd

# 1. 2025년 전기·전자 101개 기업 데이터 필터링
df_elec_2025 = df[(df['평가년도'] == 2025) & (df['업종명'] == '전기·전자')].copy()

# 2. 한국표준산업분류(KSIC) C26 vs C28 분류 함수 정의
def assign_ksic(company_name):
    company_name = str(company_name).strip()

    # C28: 전기장비 제조업 (배터리/이차전지, 전선, 발전기, 조명, 가전제품, 전력기기 등)
    c28_keywords = [
        'LG에너지솔루션', '삼성SDI', 'SK아이이테크놀로지', '엘앤에프', '포스코퓨처엠', '에코프로비엠', 'DN오토모티브', 'LS머트리얼즈', '모베이스전자', '현대모비스', 'HL만도',
        'HD현대일렉트릭', 'LS일렉트릭', 'LS전선', '대한전선', '가온전선', '일진전기', '효성중공업', '제룡전기', '광명전기', '보성파워텍', '서전기전', 'HD현대에너지솔루션', 'KH 필룩스',
        'LG전자', '위니아', '쿠쿠홈시스', '쿠쿠홀딩스', '신일전자', '파セ', '오텍', '위닉스', 'PN풍년', '두산퓨얼셀'
    ]

    # C28 기업 리스트에 해당하는 키워드가 기업명에 포함되어 있으면 C28로 지정
    if any(k in company_name for k in c28_keywords):
        return 'C28: 전기장비 제조업'
    else:
        return 'C26: 전자부품, 컴퓨터, 영상, 음향 및 통신장비 제조업'

# 3. 분류 적용
df_elec_2025['KSIC_분류'] = df_elec_2025['기업명'].apply(assign_ksic)

# 4. 분류별 기업 수 집계 및 출력
ksic_summary = df_elec_2025.groupby('KSIC_분류')['기업명'].agg(
    기업수='nunique',
    기업목록=lambda x: ", ".join(x)
).reset_index()

print("=" * 80)
print("■ 1. 한국표준산업분류(C26 / C28) 대분류별 기업 수 집계")
print("=" * 80)
print(ksic_summary[['KSIC_분류', '기업수']].to_string(index=False))
print(f"\n총 포함 기업 수: {ksic_summary['기업수'].sum()}개사")

print("\n" + "=" * 80)
print("■ 2. 세부 분류별 포함 기업 목록")
print("=" * 80)
for _, row in ksic_summary.iterrows():
    print(f"\n▶ [{row['KSIC_분류']}] - 총 {row['기업수']}개사")
    print(f"   기업 목록: {row['기업목록']}")

■ 1. 한국표준산업분류(C26 / C28) 대분류별 기업 수 집계
                          KSIC_분류  기업수
C26: 전자부품, 컴퓨터, 영상, 음향 및 통신장비 제조업   82
                    C28: 전기장비 제조업   19

총 포함 기업 수: 101개사

■ 2. 세부 분류별 포함 기업 목록

▶ [C26: 전자부품, 컴퓨터, 영상, 음향 및 통신장비 제조업] - 총 82개사
   기업 목록: 에스앤에스텍, LX세미콘, 주성코퍼레이션, 하나머티리얼즈, 서진시스템, 유티아이, 인텔리안테크, 드림텍, 해성디에스, 제이앤티씨, 덕산네오룩스, RFHIC, 심텍, 솔루엠, 한화시스템, 이녹스첨단소재, 솔루스첨단소재, 대덕전자, 더블유씨피, 가온칩스, 에코프로머티, SK하이닉스, DB하이텍, 금호전기, 삼화콘덴서공업, 세방전지, 한솔테크닉스, 써니전자, 씨아이테크, 삼영전자공업, 휴니드테크놀러지스, 삼성전자, 대원전선, 선도전기, 이수페타시스, 코리아써키트, 대동전자, 아남전자, 경인전자, 삼성전기, 경동나비엔, 삼화전기, 엘에스일렉트릭, LG이노텍, 삼화전자공업, 와이투솔루션, 신성이엔지, 성문전자, 에이엔피, 광전자, 티에이치엔, 엑시큐어하이트론, 롯데에너지머티리얼즈, 일진디스플, 시노펙스, 한국단자공업, 신도리코, 케이엠더블유, 시그네틱스, 자화전자, 네패스, LG디스플레이, SFA반도체, 에프에스티, 주연테크, 서울반도체, 인탑스, 인터플렉스, 리노공업, 에스피지, 신성델타테크, 하나마이크론, 다원시스, 대주전자재료, 제주반도체, 아이티엠반도체, 비에이치, 이엠텍, 파트론, KEC, 칩스앤미디어, 엠씨넥스

▶ [C28: 전기장비 제조업] - 총 19개사
   기업 목록: 일진전기, 에코프로비엠, HD현대일렉트릭, 효성중공업, HD현대에너지솔루션, 두산퓨얼셀, SK아이이테크놀로지, LG에너지솔루션, LS머트리얼즈, 가온전선, 대한전선, 포스코퓨처엠, 삼성SDI, DN오토모티브, 광명전기, 제룡전기, KH 필룩

In [ ]:
# 현재 데이터프레임의 컬럼 이름 전체 출력
print(df.columns.tolist())

['기업명', '기업코드', '평가년도', 'ESG등급', '환경', '사회', '지배구조', 'ESG 등급조정', '환경 등급조정', '사회 등급조정', '지배구조 등급조정', '시장구분', '업종명', '종가', '대비', '등락률', '시가총액', '지속가능경영보고서_발행', '기업지배구조보고서_발행', '자산총액']


In [ ]:
# C26 데이터셋 내 순수 반도체 밸류체인 33개사 전체 목록
semi_33_list = [
    # 1. IDM 및 메모리/파운드리/팹리스
    '삼성전자', 'SK하이닉스', 'DB하이텍', 'LX세미콘', '가온칩스',
    '제주반도체', '칩스앤미디어', 'KEC', '텔레칩스', '어보브반도체',

    # 2. 후공정 및 패키징/테스트 (OSAT)
    '하나마이크론', 'SFA반도체', '네패스', '시그네틱스', '두산테스나',

    # 3. 반도체 소재 및 부품
    '에스앤에스텍', '하나머티리얼즈', '리노공업', '해성디에스', '덕산네오룩스',
    '대주전자재료', 'TCK', '원익QnC', '솔브레인', '동진쎄미켐',

    # 4. 반도체 장비 및 공정 기기
    '주성코퍼레이션', '에프에스티', '한미반도체', '원익IPS', '유진테크',
    '피에스케이', '이오테크닉스', '케이씨텍'
]

# 보유하신 데이터프레임(df)에서 33개사 필터링
df_semi = df[df['기업명'].isin(semi_33_list)].copy()
semiconductor_companies = df_semi['기업명'].unique().tolist()

print(f"추출된 반도체 기업 수: {len(semiconductor_companies)}개사")
print(semiconductor_companies)

추출된 반도체 기업 수: 20개사
['에스앤에스텍', 'LX세미콘', '주성코퍼레이션', '하나머티리얼즈', '해성디에스', '덕산네오룩스', '가온칩스', 'SK하이닉스', 'DB하이텍', '삼성전자', '시그네틱스', '네패스', 'SFA반도체', '에프에스티', '리노공업', '하나마이크론', '대주전자재료', '제주반도체', 'KEC', '칩스앤미디어']


================================================================================
■ [코드 281 하위 4자리 기준] 기업명 추출 결과
================================================================================

**[2811. 전동기, 발전기 및 전기 변환장치 제조업] - 총 7개사**


   기업 목록: HD현대에너지솔루션, HD현대일렉트릭, 두산퓨얼셀, 솔루엠, 와이투솔루션, 제룡전기, 효성중공업

**[2812. 전기 공급 및 제어장치 제조업] - 총 4개사**


   기업 목록: 가온전선, 광명전기, 선도전기, 엘에스일렉트릭



* **C26에 포함된 하위 기업 (총 4개사)**
* **2811**: 솔루엠, 와이투솔루션
* **2812**: 선도전기, 엘에스일렉트릭


* **C28에 포함된 하위 기업 (총 7개사)**
* **2811**: HD현대에너지솔루션, HD현대일렉트릭, 두산퓨얼셀, 제룡전기, 효성중공업
* **2812**: 가온전선, 광명전기


### C26 내부 구성 (세부 산업 분류)

C26은 하위에 다음과 같이 다양한 세부 분야를 포함하고 있습니다.

1. **반도체 및 반도체 부품/장비 (C261)**
* 삼성전자, SK하이닉스, DB하이텍, LX세미콘, 리노공업, 하나마이크론, SFA반도체 등


2. **전자부품 (디스플레이, PCB, 기타 부품) (C262)**
* LG디스플레이, 삼성전기, LG이노텍, 심텍, 대덕전자, 코리아써키트 등


3. **컴퓨터 및 주변장치 (C263)**
* 주연테크, 기타 컴퓨팅 기기 제조사


4. **통신 및 방송 장비 (C264)**
* 한화시스템, RFHIC, 케이엠더블유, 인텔리안테크 등


5. **영상 및 음향 기기 (C265)**
* 가전/통신 관련 영상·음향 부품 제조업체



---

### 요약하자면

* 반도체는 C26의 '핵심 부분집합'입니다. (삼성전자, SK하이닉스 등)
* 하지만 C26 안에는 반도체 외에도 **디스플레이, 기판(PCB), 통신장비, 카메라 모듈 등 다양한 전자부품 제조업체**가 함께 묶여 있습니다.

##C26 분석

In [ ]:
import pandas as pd

# 1. C26 대상 기업 82개 목록
c26_companies = [
    '에스앤에스텍', 'LX세미콘', '주성코퍼레이션', '하나머티리얼즈', '서진시스템', '유티아이', '인텔리안테크', '드림텍',
    '해성디에스', '제이앤티씨', '덕산네오룩스', 'RFHIC', '심텍', '솔루엠', '한화시스템', '이녹스첨단소재',
    '솔루스첨단소재', '대덕전자', '더블유씨피', '가온칩스', '에코프로머티', 'SK하이닉스', 'DB하이텍', '금호전기',
    '삼화콘덴서공업', '세방전지', '한솔테크닉스', '써니전자', '씨아이테크', '삼영전자공업', '휴니드테크놀러지스', '삼성전자',
    '대원전선', '선도전기', '이수페타시스', '코리아써키트', '대동전자', '아남전자', '경인전자', '삼성전기',
    '경동나비엔', '삼화전기', '엘에스일렉트릭', 'LG이노텍', '삼화전자공업', '와이투솔루션', '신성이엔지', '성문전자',
    '에이엔피', '광전자', '티에이치엔', '엑시큐어하이트론', '롯데에너지머티리얼즈', '일진디스플', '시노펙스', '한국단자공업',
    '신도리코', '케이엠더블유', '시그네틱스', '자화전자', '네패스', 'LG디스플레이', 'SFA반도체', '에프에스티',
    '주연테크', '서울반도체', '인탑스', '인터플렉스', '리노공업', '에스피지', '신성델타테크', '하나마이크론',
    '다원시스', '대주전자재료', '제주반도체', '아이티엠반도체', '비에이치', '이엠텍', '파트론', 'KEC', '칩스앤미디어', '엠씨넥스'
]

# 2. 2022~2025년 데이터 추출 및 연도별 존재 여부 피벗 생성
df_c26 = df[(df['기업명'].isin(c26_companies)) & (df['평가년도'].between(2022, 2025))].copy()
pv = pd.pivot_table(df_c26, index='기업명', columns='평가년도', values='ESG등급', aggfunc='count').fillna(0).astype(int)

# 누락 기업 채우기
for comp in c26_companies:
    if comp not in pv.index:
        pv.loc[comp] = [0, 0, 0, 0]
pv.columns = [2022, 2023, 2024, 2025]

# ----------------------------------------------------
# 3. 2022~2025 편출입 정밀 분석
# ----------------------------------------------------
full_4yrs = pv[(pv[2022]==1) & (pv[2023]==1) & (pv[2024]==1) & (pv[2025]==1)].index.tolist()
excluded_any = pv[~( (pv[2022]==1) & (pv[2023]==1) & (pv[2024]==1) & (pv[2025]==1) )].index.tolist()

# 재편승(재편입) 정밀 판별: 과거 평가 대상(1)이었다가 제외(0)된 후, 다시 평가 대상(1)이 된 경우
reincluded = []
for comp in excluded_any:
    history = list(pv.loc[comp]) # 예: [1, 0, 1, 1]
    # 1 -> 0 -> 1 패턴 검사
    for i in range(len(history) - 2):
        if history[i] == 1 and history[i+1] == 0 and any(h == 1 for h in history[i+2:]):
            reincluded.append(comp)
            break

print("=" * 80)
print(f"■ C26 대상 기업(총 {len(c26_companies)}개사) 2022~2025 평가년도 검수 결과")
print("=" * 80)
print(f"1. 4개년 연속 완벽 수록 기업: {len(full_4yrs)}개사")
print(f"2. 4개년 중 한번이라도 제외/누락/신규편입된 기업: {len(excluded_any)}개사")
print(f"3. 평가 대상 제외 후 다시 편승(재편입)한 기업: {len(reincluded)}개사 ({', '.join(reincluded) if reincluded else '없음'})")

■ C26 대상 기업(총 82개사) 2022~2025 평가년도 검수 결과
1. 4개년 연속 완벽 수록 기업: 65개사
2. 4개년 중 한번이라도 제외/누락/신규편입된 기업: 17개사
3. 평가 대상 제외 후 다시 편승(재편입)한 기업: 1개사 (제이앤티씨)


In [ ]:
import pandas as pd

# 1. C26 대상 기업 82개 목록
c26_companies = [
    '에스앤에스텍', 'LX세미콘', '주성코퍼레이션', '하나머티리얼즈', '서진시스템', '유티아이', '인텔리안테크', '드림텍',
    '해성디에스', '제이앤티씨', '덕산네오룩스', 'RFHIC', '심텍', '솔루엠', '한화시스템', '이녹스첨단소재',
    '솔루스첨단소재', '대덕전자', '더블유씨피', '가온칩스', '에코프로머티', 'SK하이닉스', 'DB하이텍', '금호전기',
    '삼화콘덴서공업', '세방전지', '한솔테크닉스', '써니전자', '씨아이테크', '삼영전자공업', '휴니드테크놀러지스', '삼성전자',
    '대원전선', '선도전기', '이수페타시스', '코리아써키트', '대동전자', '아남전자', '경인전자', '삼성전기',
    '경동나비엔', '삼화전기', '엘에스일렉트릭', 'LG이노텍', '삼화전자공업', '와이투솔루션', '신성이엔지', '성문전자',
    '에이엔피', '광전자', '티에이치엔', '엑시큐어하이트론', '롯데에너지머티리얼즈', '일진디스플', '시노펙스', '한국단자공업',
    '신도리코', '케이엠더블유', '시그네틱스', '자화전자', '네패스', 'LG디스플레이', 'SFA반도체', '에프에스티',
    '주연테크', '서울반도체', '인탑스', '인터플렉스', '리노공업', '에스피지', '신성델타테크', '하나마이크론',
    '다원시스', '대주전자재료', '제주반도체', '아이티엠반도체', '비에이치', '이엠텍', '파트론', 'KEC', '칩스앤미디어', '엠씨넥스'
]

# 2. 2022~2025년 데이터 추출 및 연도별 존재 여부 피벗 생성
df_c26 = df[(df['기업명'].isin(c26_companies)) & (df['평가년도'].between(2022, 2025))].copy()
pv = pd.pivot_table(df_c26, index='기업명', columns='평가년도', values='ESG등급', aggfunc='count').fillna(0).astype(int)

# 누락 기업 채우기
for comp in c26_companies:
    if comp not in pv.index:
        pv.loc[comp] = [0, 0, 0, 0]
pv.columns = [2022, 2023, 2024, 2025]

# 3. 구분 로직
full_4yrs = pv[(pv[2022]==1) & (pv[2023]==1) & (pv[2024]==1) & (pv[2025]==1)].index.tolist()
excluded_any = pv[~( (pv[2022]==1) & (pv[2023]==1) & (pv[2024]==1) & (pv[2025]==1) )].index.tolist()

# 재편승(재편입) 판별 (1 -> 0 -> 1 패턴)
reincluded = []
for comp in excluded_any:
    history = list(pv.loc[comp])
    for i in range(len(history) - 2):
        if history[i] == 1 and history[i+1] == 0 and any(h == 1 for h in history[i+2:]):
            reincluded.append(comp)
            break

# 2025년에만 존재하는 기업 (2022, 2023, 2024년은 0이고 2025년만 1인 기업)
only_2025 = pv[(pv[2022]==0) & (pv[2023]==0) & (pv[2024]==0) & (pv[2025]==1)].index.tolist()

# 4. 결과 출력
print("=" * 80)
print(f"■ C26 대상 기업(총 {len(c26_companies)}개사) 2022~2025 평가년도 검수 결과")
print("=" * 80)
print(f"1. 4개년 연속 완벽 수록 기업: {len(full_4yrs)}개사")
print(f"2. 4개년 중 한번이라도 제외/누락/신규편입된 기업: {len(excluded_any)}개사")

# 2번 대상 기업들의 4개년치 (0, 1) 현황표 출력
if excluded_any:
    print("\n   [상세 누락 / 신규 편입 연도 현황 (1: 포함, 0: 미포함)]")
    print("-" * 55)
    print(pv.loc[excluded_any].to_string())
    print("-" * 55)

print(f"\n3. 평가 대상 제외 후 다시 편승(재편입)한 기업: {len(reincluded)}개사 ({', '.join(reincluded) if reincluded else '없음'})")

print(f"\n4. 2025년에만 존재하는 기업 (신규 편입): {len(only_2025)}개사")
if only_2025:
    print(f"   기업 목록: {', '.join(only_2025)}")

■ C26 대상 기업(총 82개사) 2022~2025 평가년도 검수 결과
1. 4개년 연속 완벽 수록 기업: 65개사
2. 4개년 중 한번이라도 제외/누락/신규편입된 기업: 17개사

   [상세 누락 / 신규 편입 연도 현황 (1: 포함, 0: 미포함)]
-------------------------------------------------------
            2022  2023  2024  2025
기업명                               
가온칩스           0     0     0     1
더블유씨피          0     0     1     1
롯데에너지머티리얼즈     0     1     1     1
솔루엠            0     1     1     1
시노펙스           0     0     0     1
신성델타테크         0     0     1     1
에스피지           0     0     1     1
에코프로머티         0     0     0     1
엑시큐어하이트론       0     0     1     1
엠씨넥스           0     1     1     1
유티아이           0     0     0     1
이엠텍            0     0     1     1
인텔리안테크         0     1     1     1
제이앤티씨          1     0     1     1
제주반도체          0     0     0     1
주성코퍼레이션        0     1     1     1
칩스앤미디어         0     0     0     1
-------------------------------------------------------

3. 평가 대상 제외 후 다시 편승(재편입)한 기업: 1개사 (제이앤티씨)

4. 2025년에만 존재하는 기업 (신규 편입): 6개사
   

In [ ]:
# ==============================================================================
# C26 업종 보고서 발행 현황 분석 (2025년 신규 편입 기업 제외 모수 76개사 기준)
# ==============================================================================

# 1. C26 82개 목록 설정
c26_companies = [
    '에스앤에스텍', 'LX세미콘', '주성코퍼레이션', '하나머티리얼즈', '서진시스템', '유티아이', '인텔리안테크', '드림텍',
    '해성디에스', '제이앤티씨', '덕산네오룩스', 'RFHIC', '심텍', '솔루엠', '한화시스템', '이녹스첨단소재',
    '솔루스첨단소재', '대덕전자', '더블유씨피', '가온칩스', '에코프로머티', 'SK하이닉스', 'DB하이텍', '금호전기',
    '삼화콘덴서공업', '세방전지', '한솔테크닉스', '써니전자', '씨아이테크', '삼영전자공업', '휴니드테크놀러지스', '삼성전자',
    '대원전선', '선도전기', '이수페타시스', '코리아써키트', '대동전자', '아남전자', '경인전자', '삼성전기',
    '경동나비엔', '삼화전기', '엘에스일렉트릭', 'LG이노텍', '삼화전자공업', '와이투솔루션', '신성이엔지', '성문전자',
    '에이엔피', '광전자', '티에이치엔', '엑시큐어하이트론', '롯데에너지머티리얼즈', '일진디스플', '시노펙스', '한국단자공업',
    '신도리코', '케이엠더블유', '시그네틱스', '자화전자', '네패스', 'LG디스플레이', 'SFA반도체', '에프에스티',
    '주연테크', '서울반도체', '인탑스', '인터플렉스', '리노공업', '에스피지', '신성델타테크', '하나마이크론',
    '다원시스', '대주전자재료', '제주반도체', '아이티엠반도체', '비에이치', '이엠텍', '파트론', 'KEC', '칩스앤미디어', '엠씨넥스'
]

# 2. 2025년 신규 편입 기업 6개사 제외 처리
df_c26 = df[(df['기업명'].isin(c26_companies)) & (df['평가년도'].between(2022, 2025))].copy()
pv = pd.pivot_table(df_c26, index='기업명', columns='평가년도', values='ESG등급', aggfunc='count').fillna(0).astype(int)
only_2025_comps = pv[(pv[2022]==0) & (pv[2023]==0) & (pv[2024]==0) & (pv[2025]==1)].index.tolist()

# 최종 분석 대상 76개사 명단
target_76_comps = [c for c in c26_companies if c not in only_2025_comps]

# 3. 2025년 최신 데이터 기준 76개사의 보고서 발행 현황 단일 카운트
df_2025 = df[(df['기업명'].isin(target_76_comps)) & (df['평가년도'] == 2025)].copy()

# 발행 유무 판단 (Y/N 처리)
is_sus_y = df_2025['지속가능경영보고서_발행'] == 'Y'
is_cg_y = df_2025['기업지배구조보고서_발행'] == 'Y'

# 1) 둘 다 발행 (Both Y)
both_yes = df_2025[is_sus_y & is_cg_y]['기업명'].tolist()

# 2) 둘 중 하나만 발행 (One Y)
one_yes = df_2025[(is_sus_y & ~is_cg_y) | (~is_sus_y & is_cg_y)]['기업명'].tolist()

# 3) 둘 다 미발행 (Both N)
both_no = df_2025[~is_sus_y & ~is_cg_y]['기업명'].tolist()

# ------------------------------------------------------------------------------
# 결과 출력
# ------------------------------------------------------------------------------
print("=" * 70)
print("■ C26 업종 보고서 발행 현황 분석 (2025년 신규 편입 기업 제외 기준)")
print("=" * 70)
print(f"• C26 전체 기업 수: {len(c26_companies)}개사")
print(f"• 2025년에만 등급이 존재하는 기업: {len(only_2025_comps)}개사 ({', '.join(only_2025_comps)})")
print(f"• 분석 대상 기업 수 (제외 후): {len(target_76_comps)}개사\n")

print(f"1. 25년 기준 지속가능경영보고서 & 기업지배구조보고서 [둘 다 발행] 기업: 총 {len(both_yes)}개사")
print(f"   기업 목록: {', '.join(both_yes)}\n")

print(f"2. 25년 기준 두 보고서 중 [하나만 발행]한 기업: 총 {len(one_yes)}개사")
print(f"   기업 목록: {', '.join(one_yes)}\n")

print(f"3. 25년 기준 두 보고서 [모두 미발행] 기업: 총 {len(both_no)}개사")
print(f"   기업 목록: {', '.join(both_no)}\n")

# 검증 출력
print("=" * 70)
print(f"※ 검증 (합계): {len(both_yes)} + {len(one_yes)} + {len(both_no)} = {len(both_yes) + len(one_yes) + len(both_no)}개사 (목표: 76개사)")
print("=" * 70)

■ C26 업종 보고서 발행 현황 분석 (2025년 신규 편입 기업 제외 기준)
• C26 전체 기업 수: 82개사
• 2025년에만 등급이 존재하는 기업: 6개사 (가온칩스, 시노펙스, 에코프로머티, 유티아이, 제주반도체, 칩스앤미디어)
• 분석 대상 기업 수 (제외 후): 76개사

1. 25년 기준 지속가능경영보고서 & 기업지배구조보고서 [둘 다 발행] 기업: 총 14개사
   기업 목록: LX세미콘, 드림텍, 해성디에스, 솔루엠, 한화시스템, SK하이닉스, DB하이텍, 세방전지, 삼성전자, 삼성전기, 엘에스일렉트릭, LG이노텍, 한국단자공업, LG디스플레이

2. 25년 기준 두 보고서 중 [하나만 발행]한 기업: 총 13개사
   기업 목록: 솔루스첨단소재, 대덕전자, 한솔테크닉스, 삼영전자공업, 이수페타시스, 코리아써키트, 경동나비엔, 신성이엔지, 롯데에너지머티리얼즈, 신도리코, 자화전자, 비에이치, 엠씨넥스

3. 25년 기준 두 보고서 [모두 미발행] 기업: 총 49개사
   기업 목록: 에스앤에스텍, 주성코퍼레이션, 하나머티리얼즈, 서진시스템, 인텔리안테크, 제이앤티씨, 덕산네오룩스, RFHIC, 심텍, 이녹스첨단소재, 더블유씨피, 금호전기, 삼화콘덴서공업, 써니전자, 씨아이테크, 휴니드테크놀러지스, 대원전선, 선도전기, 대동전자, 아남전자, 경인전자, 삼화전기, 삼화전자공업, 와이투솔루션, 성문전자, 에이엔피, 광전자, 티에이치엔, 엑시큐어하이트론, 일진디스플, 케이엠더블유, 시그네틱스, 네패스, SFA반도체, 에프에스티, 주연테크, 서울반도체, 인탑스, 인터플렉스, 리노공업, 에스피지, 신성델타테크, 하나마이크론, 다원시스, 대주전자재료, 아이티엠반도체, 이엠텍, 파트론, KEC

※ 검증 (합계): 14 + 13 + 49 = 76개사 (목표: 76개사)


##1.보고서 [둘 다 발행] 기업: 총 14개사


   기업 목록: LX세미콘, 드림텍, 해성디에스, 솔루엠, 한화시스템, SK하이닉스, DB하이텍, 세방전지, 삼성전자, 삼성전기, 엘에스일렉트릭, LG이노텍, 한국단자공업, LG디스플레이


In [ ]:
import pandas as pd
import plotly.graph_objects as go

# ----------------------------------------------------
# 1. 대상 기업(둘 다 발행 기업 19개사) 데이터 필터링
# ----------------------------------------------------
c26_companies = [
    '에스앤에스텍', 'LX세미콘', '주성코퍼레이션', '하나머티리얼즈', '서진시스템', '유티아이', '인텔리안테크', '드림텍',
    '해성디에스', '제이앤티씨', '덕산네오룩스', 'RFHIC', '심텍', '솔루엠', '한화시스템', '이녹스첨단소재',
    '솔루스첨단소재', '대덕전자', '더블유씨피', '가온칩스', '에코프로머티', 'SK하이닉스', 'DB하이텍', '금호전기',
    '삼화콘덴서공업', '세방전지', '한솔테크닉스', '써니전자', '씨아이테크', '삼영전자공업', '휴니드테크놀러지스', '삼성전자',
    '대원전선', '선도전기', '이수페타시스', '코리아써키트', '대동전자', '아남전자', '경인전자', '삼성전기',
    '경동나비엔', '삼화전기', '엘에스일렉트릭', 'LG이노텍', '삼화전자공업', '와이투솔루션', '신성이엔지', '성문전자',
    '에이엔피', '광전자', '티에이치엔', '엑시큐어하이트론', '롯데에너지머티리얼즈', '일진디스플', '시노펙스', '한국단자공업',
    '신도리코', '케이엠더블유', '시그네틱스', '자화전자', '네패스', 'LG디스플레이', 'SFA반도체', '에프에스티',
    '주연테크', '서울반도체', '인탑스', '인터플렉스', '리노공업', '에스피지', '신성델타테크', '하나마이크론',
    '다원시스', '대주전자재료', '제주반도체', '아이티엠반도체', '비에이치', '이엠텍', '파트론', 'KEC', '칩스앤미디어', '엠씨넥스'
]

df_c26 = df[(df['기업명'].isin(c26_companies)) & (df['평가년도'].between(2022, 2025))].copy()
pv = pd.pivot_table(df_c26, index='기업명', columns='평가년도', values='ESG등급', aggfunc='count').fillna(0).astype(int)
only_2025_comps = pv[(pv[2022]==0) & (pv[2023]==0) & (pv[2024]==0) & (pv[2025]==1)].index.tolist()
target_76_comps = [c for c in c26_companies if c not in only_2025_comps]

df_2025_target = df[(df['기업명'].isin(target_76_comps)) & (df['평가년도'] == 2025)]
both_yes_comps = df_2025_target[
    (df_2025_target['지속가능경영보고서_발행'] == 'Y') &
    (df_2025_target['기업지배구조보고서_발행'] == 'Y')
]['기업명'].tolist()

df_plot = df[(df['기업명'].isin(both_yes_comps)) & (df['평가년도'].between(2022, 2025))].copy()

# ----------------------------------------------------
# 2. 등급 수치화 및 정렬용 피벗 생성 (오류 수정 구간)
# ----------------------------------------------------
grade_map = {'S': 7, 'A+': 6, 'A': 5, 'B+': 4, 'B': 3, 'C': 2, 'D': 1}
df_plot['ESG등급_num'] = df_plot['ESG등급'].map(grade_map).fillna(0)

# 기초 피벗 테이블 먼저 구성
pv_num = df_plot.pivot_table(index='기업명', columns='평가년도', values='ESG등급_num', aggfunc='first').fillna(0)

# 정렬용 지표 계산 (A 이상 횟수, 총점수, 2025년 점수)
pv_num['A이상횟수'] = (pv_num >= 5).sum(axis=1)
pv_num['총점수'] = pv_num[[2022, 2023, 2024, 2025]].sum(axis=1)
pv_num['2025점수'] = pv_num[2025]

# 내림차순 정렬 기준 적용
sort_order = pv_num.sort_values(
    by=['A이상횟수', '총점수', '2025점수'],
    ascending=[False, False, False]
).index.tolist()

# 최종 출력용 피벗 테이블 재정렬
pivot_esg = df_plot.pivot(index='기업명', columns='평가년도', values='ESG등급').reindex(sort_order).fillna('-')
pivot_esg_num = df_plot.pivot(index='기업명', columns='평가년도', values='ESG등급_num').reindex(sort_order).fillna(0)
pivot_env = df_plot.pivot(index='기업명', columns='평가년도', values='환경').reindex(sort_order).fillna('-')
pivot_soc = df_plot.pivot(index='기업명', columns='평가년도', values='사회').reindex(sort_order).fillna('-')
pivot_gov = df_plot.pivot(index='기업명', columns='평가년도', values='지배구조').reindex(sort_order).fillna('-')

pivot_sus = df_plot.pivot(index='기업명', columns='평가년도', values='지속가능경영보고서_발행').reindex(sort_order).fillna('N')
pivot_cg = df_plot.pivot(index='기업명', columns='평가년도', values='기업지배구조보고서_발행').reindex(sort_order).fillna('N')

# Hover 텍스트 작성
hover_text = []
for comp in pivot_esg.index:
    row_text = []
    for yr in pivot_esg.columns:
        esg = pivot_esg.loc[comp, yr]
        e = pivot_env.loc[comp, yr]
        s = pivot_soc.loc[comp, yr]
        g = pivot_gov.loc[comp, yr]
        sus = pivot_sus.loc[comp, yr]
        cg = pivot_cg.loc[comp, yr]

        txt = (f"<b>{comp} ({yr}년)</b><br>"
               f"--------------------<br>"
               f"• ESG 종합등급: {esg}<br>"
               f"• 환경(E): {e} | 사회(S): {s} | 지배구조(G): {g}<br>"
               f"• 지속가능경영보고서: {sus}<br>"
               f"• 기업지배구조보고서: {cg}")
        row_text.append(txt)
    hover_text.append(row_text)

# ----------------------------------------------------
# 3. 블루 계열 Heatmap 시각화
# ----------------------------------------------------
blue_colorscale = [
    [0.0, "rgba(220, 225, 230, 0.4)"], # D 이하
    [0.2, "rgba(189, 215, 238, 0.5)"], # C
    [0.4, "rgba(155, 194, 230, 0.7)"], # B
    [0.6, "rgba(91, 155, 213, 0.85)"], # B+
    [0.8, "rgba(47, 117, 181, 0.95)"], # A
    [1.0, "rgba(31, 78, 121, 1.0)"]    # A+ / S
]

fig = go.Figure(data=go.Heatmap(
    z=pivot_esg_num.values,
    x=[f"{yr}년" for yr in pivot_esg.columns],
    y=pivot_esg.index,
    text=pivot_esg.values,
    texttemplate="<b>%{text}</b>",
    textfont={"size": 13, "color": "white"},
    hoverinfo="text",
    hovertext=hover_text,
    colorscale=blue_colorscale,
    showscale=False
))

fig.update_layout(
    title={
        'text': "<b>C26 보고서 [둘 다 발행] 19개 기업 연도별 ESG 등급 현황 (4개년 누적 A등급 이상 우선순위)</b><br><sup>* 4개년 동안 A 이상 등급을 꾸준히 유지한 기업부터 상단 배치</sup>",
        'x': 0.5,
        'xanchor': 'center'
    },
    xaxis=dict(title="<b>평가년도</b>", tickfont=dict(size=12)),
    yaxis=dict(title="<b>기업명 (우수 기업 순)</b>", autorange="reversed"),
    font=dict(
        family="NanumGothic, Apple SD Gothic Neo, Malgun Gothic, sans-serif",
        size=12,
        color="#333333"
    ),
    width=900,
    height=800,
    margin=dict(l=150, r=50, t=100, b=50),
    plot_bgcolor="white"
)

# 코랩 렌더링 오류 방지
try:
    fig.show(renderer="colab")
except:
    fig.show()

##2. 보고서 [하나만 발행]한 기업: 총 13개사
   기업 목록: 솔루스첨단소재, 대덕전자, 한솔테크닉스, 삼영전자공업, 이수페타시스, 코리아써키트, 경동나비엔, 신성이엔지, 롯데에너지머티리얼즈, 신도리코, 자화전자, 비에이치, 엠씨넥스


In [ ]:
import pandas as pd
import plotly.graph_objects as go

# ----------------------------------------------------
# 1. 대상 기업([하나만 발행] 기업 13개사) 데이터 필터링
# ----------------------------------------------------
c26_companies = [
    '에스앤에스텍', 'LX세미콘', '주성코퍼레이션', '하나머티리얼즈', '서진시스템', '유티아이', '인텔리안테크', '드림텍',
    '해성디에스', '제이앤티씨', '덕산네오룩스', 'RFHIC', '심텍', '솔루엠', '한화시스템', '이녹스첨단소재',
    '솔루스첨단소재', '대덕전자', '더블유씨피', '가온칩스', '에코프로머티', 'SK하이닉스', 'DB하이텍', '금호전기',
    '삼화콘덴서공업', '세방전지', '한솔테크닉스', '써니전자', '씨아이테크', '삼영전자공업', '휴니드테크놀러지스', '삼성전자',
    '대원전선', '선도전기', '이수페타시스', '코리아써키트', '대동전자', '아남전자', '경인전자', '삼성전기',
    '경동나비엔', '삼화전기', '엘에스일렉트릭', 'LG이노텍', '삼화전자공업', '와이투솔루션', '신성이엔지', '성문전자',
    '에이엔피', '광전자', '티에이치엔', '엑시큐어하이트론', '롯데에너지머티리얼즈', '일진디스플', '시노펙스', '한국단자공업',
    '신도리코', '케이엠더블유', '시그네틱스', '자화전자', '네패스', 'LG디스플레이', 'SFA반도체', '에프에스티',
    '주연테크', '서울반도체', '인탑스', '인터플렉스', '리노공업', '에스피지', '신성델타테크', '하나마이크론',
    '다원시스', '대주전자재료', '제주반도체', '아이티엠반도체', '비에이치', '이엠텍', '파트론', 'KEC', '칩스앤미디어', '엠씨넥스'
]

# 2025년 신규 편입 기업 제외 모수
df_c26 = df[(df['기업명'].isin(c26_companies)) & (df['평가년도'].between(2022, 2025))].copy()
pv = pd.pivot_table(df_c26, index='기업명', columns='평가년도', values='ESG등급', aggfunc='count').fillna(0).astype(int)
only_2025_comps = pv[(pv[2022]==0) & (pv[2023]==0) & (pv[2024]==0) & (pv[2025]==1)].index.tolist()
target_76_comps = [c for c in c26_companies if c not in only_2025_comps]

# 2025년 기준 [하나만 발행]한 기업 13개사 추출
df_2025_target = df[(df['기업명'].isin(target_76_comps)) & (df['평가년도'] == 2025)]
is_sus_y = df_2025_target['지속가능경영보고서_발행'] == 'Y'
is_cg_y = df_2025_target['기업지배구조보고서_발행'] == 'Y'

one_yes_comps = df_2025_target[(is_sus_y & ~is_cg_y) | (~is_sus_y & is_cg_y)]['기업명'].tolist()

df_plot = df[(df['기업명'].isin(one_yes_comps)) & (df['평가년도'].between(2022, 2025))].copy()

# ----------------------------------------------------
# 2. 등급 수치화 및 정렬용 피벗 생성 (누적 A등급 우선순위)
# ----------------------------------------------------
grade_map = {'S': 7, 'A+': 6, 'A': 5, 'B+': 4, 'B': 3, 'C': 2, 'D': 1}
df_plot['ESG등급_num'] = df_plot['ESG등급'].map(grade_map).fillna(0)

# 기초 피벗 테이블 구성
pv_num = df_plot.pivot_table(index='기업명', columns='평가년도', values='ESG등급_num', aggfunc='first').fillna(0)

# 정렬 지표 계산
pv_num['A이상횟수'] = (pv_num >= 5).sum(axis=1)
pv_num['총점수'] = pv_num[[2022, 2023, 2024, 2025]].sum(axis=1)
pv_num['2025점수'] = pv_num[2025]

# 내림차순 정렬 기준 적용
sort_order = pv_num.sort_values(
    by=['A이상횟수', '총점수', '2025점수'],
    ascending=[False, False, False]
).index.tolist()

# 최종 출력용 피벗 테이블 재정렬
pivot_esg = df_plot.pivot(index='기업명', columns='평가년도', values='ESG등급').reindex(sort_order).fillna('-')
pivot_esg_num = df_plot.pivot(index='기업명', columns='평가년도', values='ESG등급_num').reindex(sort_order).fillna(0)
pivot_env = df_plot.pivot(index='기업명', columns='평가년도', values='환경').reindex(sort_order).fillna('-')
pivot_soc = df_plot.pivot(index='기업명', columns='평가년도', values='사회').reindex(sort_order).fillna('-')
pivot_gov = df_plot.pivot(index='기업명', columns='평가년도', values='지배구조').reindex(sort_order).fillna('-')

pivot_sus = df_plot.pivot(index='기업명', columns='평가년도', values='지속가능경영보고서_발행').reindex(sort_order).fillna('N')
pivot_cg = df_plot.pivot(index='기업명', columns='평가년도', values='기업지배구조보고서_발행').reindex(sort_order).fillna('N')

# Hover 텍스트 작성
hover_text = []
for comp in pivot_esg.index:
    row_text = []
    for yr in pivot_esg.columns:
        esg = pivot_esg.loc[comp, yr]
        e = pivot_env.loc[comp, yr]
        s = pivot_soc.loc[comp, yr]
        g = pivot_gov.loc[comp, yr]
        sus = pivot_sus.loc[comp, yr]
        cg = pivot_cg.loc[comp, yr]

        txt = (f"<b>{comp} ({yr}년)</b><br>"
               f"--------------------<br>"
               f"• ESG 종합등급: {esg}<br>"
               f"• 환경(E): {e} | 사회(S): {s} | 지배구조(G): {g}<br>"
               f"• 지속가능경영보고서: {sus}<br>"
               f"• 기업지배구조보고서: {cg}")
        row_text.append(txt)
    hover_text.append(row_text)

# ----------------------------------------------------
# 3. 블루 계열 Heatmap 시각화
# ----------------------------------------------------
blue_colorscale = [
    [0.0, "rgba(220, 225, 230, 0.4)"], # D 이하
    [0.2, "rgba(189, 215, 238, 0.5)"], # C
    [0.4, "rgba(155, 194, 230, 0.7)"], # B
    [0.6, "rgba(91, 155, 213, 0.85)"], # B+
    [0.8, "rgba(47, 117, 181, 0.95)"], # A
    [1.0, "rgba(31, 78, 121, 1.0)"]    # A+ / S
]

fig = go.Figure(data=go.Heatmap(
    z=pivot_esg_num.values,
    x=[f"{yr}년" for yr in pivot_esg.columns],
    y=pivot_esg.index,
    text=pivot_esg.values,
    texttemplate="<b>%{text}</b>",
    textfont={"size": 13, "color": "white"},
    hoverinfo="text",
    hovertext=hover_text,
    colorscale=blue_colorscale,
    showscale=False
))

fig.update_layout(
    title={
        'text': "<b>C26 보고서 [하나만 발행] 13개 기업 연도별 ESG 등급 현황 (4개년 누적 A등급 이상 우선순위)</b><br><sup>* 4개년 동안 A 이상 등급을 꾸준히 유지한 기업부터 상단 배치</sup>",
        'x': 0.5,
        'xanchor': 'center'
    },
    xaxis=dict(title="<b>평가년도</b>", tickfont=dict(size=12)),
    yaxis=dict(title="<b>기업명 (우수 기업 순)</b>", autorange="reversed"),
    font=dict(
        family="NanumGothic, Apple SD Gothic Neo, Malgun Gothic, sans-serif",
        size=12,
        color="#333333"
    ),
    width=900,
    height=650,
    margin=dict(l=150, r=50, t=100, b=50),
    plot_bgcolor="white"
)

# 코랩 렌더링 오류 방지
try:
    fig.show(renderer="colab")
except:
    fig.show()

##3. 보고서 [모두 미발행] 기업: 총 49개사


   기업 목록: 에스앤에스텍, 주성코퍼레이션, 하나머티리얼즈, 서진시스템, 인텔리안테크, 제이앤티씨, 덕산네오룩스, RFHIC, 심텍, 이녹스첨단소재, 더블유씨피, 금호전기, 삼화콘덴서공업, 써니전자, 씨아이테크, 휴니드테크놀러지스, 대원전선, 선도전기, 대동전자, 아남전자, 경인전자, 삼화전기, 삼화전자공업, 와이투솔루션, 성문전자, 에이엔피, 광전자, 티에이치엔, 엑시큐어하이트론, 일진디스플, 케이엠더블유, 시그네틱스, 네패스, SFA반도체, 에프에스티, 주연테크, 서울반도체, 인탑스, 인터플렉스, 리노공업, 에스피지, 신성델타테크, 하나마이크론, 다원시스, 대주전자재료, 아이티엠반도체, 이엠텍, 파트론, KEC

In [ ]:
import pandas as pd
import plotly.express as px

# ----------------------------------------------------
# 1. 49개사 데이터 필터링
# ----------------------------------------------------
c26_companies = [
    '에스앤에스텍', 'LX세미콘', '주성코퍼레이션', '하나머티리얼즈', '서진시스템', '유티아이', '인텔리안테크', '드림텍',
    '해성디에스', '제이앤티씨', '덕산네오룩스', 'RFHIC', '심텍', '솔루엠', '한화시스템', '이녹스첨단소재',
    '솔루스첨단소재', '대덕전자', '더블유씨피', '가온칩스', '에코프로머티', 'SK하이닉스', 'DB하이텍', '금호전기',
    '삼화콘덴서공업', '세방전지', '한솔테크닉스', '써니전자', '씨아이테크', '삼영전자공업', '휴니드테크놀러지스', '삼성전자',
    '대원전선', '선도전기', '이수페타시스', '코리아써키트', '대동전자', '아남전자', '경인전자', '삼성전기',
    '경동나비엔', '삼화전기', '엘에스일렉트릭', 'LG이노텍', '삼화전자공업', '와이투솔루션', '신성이엔지', '성문전자',
    '에이엔피', '광전자', '티에이치엔', '엑시큐어하이트론', '롯데에너지머티리얼즈', '일진디스플', '시노펙스', '한국단자공업',
    '신도리코', '케이엠더블유', '시그네틱스', '자화전자', '네패스', 'LG디스플레이', 'SFA반도체', '에프에스티',
    '주연테크', '서울반도체', '인탑스', '인터플렉스', '리노공업', '에스피지', '신성델타테크', '하나마이크론',
    '다원시스', '대주전자재료', '제주반도체', '아이티엠반도체', '비에이치', '이엠텍', '파트론', 'KEC', '칩스앤미디어', '엠씨넥스'
]

# 대상 76개사 추출
df_c26 = df[(df['기업명'].isin(c26_companies)) & (df['평가년도'].between(2022, 2025))].copy()
pv = pd.pivot_table(df_c26, index='기업명', columns='평가년도', values='ESG등급', aggfunc='count').fillna(0).astype(int)
only_2025_comps = pv[(pv[2022]==0) & (pv[2023]==0) & (pv[2024]==0) & (pv[2025]==1)].index.tolist()
target_76_comps = [c for c in c26_companies if c not in only_2025_comps]

# 2025년 기준 [둘 다 미발행] 49개사
df_2025 = df[(df['기업명'].isin(target_76_comps)) & (df['평가년도'] == 2025)]
is_sus_y = df_2025['지속가능경영보고서_발행'] == 'Y'
is_cg_y = df_2025['기업지배구조보고서_발행'] == 'Y'
both_no_comps = df_2025[~is_sus_y & ~is_cg_y]['기업명'].tolist()

df_no_rep = df[(df['기업명'].isin(both_no_comps)) & (df['평가년도'].between(2022, 2025))].copy()

# ----------------------------------------------------
# 2. 연도별 ESG 등급 분포 누적 막대그래프 시각화
# ----------------------------------------------------
grade_order = ['A+', 'A', 'B+', 'B', 'C', 'D']
dist_df = df_no_rep.groupby(['평가년도', 'ESG등급']).size().reset_index(name='기업수')

fig = px.bar(
    dist_df,
    x='평가년도',
    y='기업수',
    color='ESG등급',
    category_orders={'ESG등급': grade_order},
    title="<b>C26 보고서 [모두 미발행] 49개사 연도별 ESG 등급 분포 추이</b>",
    text='기업수',
    color_discrete_sequence=px.colors.qualitative.Pastel
)

fig.update_layout(
    xaxis=dict(title="평가년도", dtick=1),
    yaxis=dict(title="기업 수 (개사)"),
    font=dict(family="NanumGothic, Apple SD Gothic Neo, Malgun Gothic, sans-serif", size=12),
    width=850,
    height=500
)

try:
    fig.show(renderer="colab")
except:
    fig.show()

# ----------------------------------------------------
# 3. 2025년 등급별 기업 리스트 요약 출력
# ----------------------------------------------------
print("=" * 70)
print("■ 2025년 기준 [모두 미발행] 49개사 등급별 기업 분류")
print("=" * 70)
df_2025_no = df_no_rep[df_no_rep['평가년도'] == 2025]

for grade in ['A+', 'A', 'B+', 'B', 'C', 'D']:
    comps = df_2025_no[df_2025_no['ESG등급'] == grade]['기업명'].tolist()
    if comps:
        print(f"• [{grade} 등급] ({len(comps)}개사): {', '.join(comps)}")

■ 2025년 기준 [모두 미발행] 49개사 등급별 기업 분류
• [A 등급] (1개사): 심텍
• [B+ 등급] (4개사): 이녹스첨단소재, 인탑스, 대주전자재료, 아이티엠반도체
• [B 등급] (1개사): 하나머티리얼즈
• [C 등급] (11개사): 에스앤에스텍, 서진시스템, 인텔리안테크, 덕산네오룩스, 삼화콘덴서공업, 네패스, SFA반도체, 에프에스티, 서울반도체, 인터플렉스, 하나마이크론
• [D 등급] (32개사): 주성코퍼레이션, 제이앤티씨, RFHIC, 더블유씨피, 금호전기, 써니전자, 씨아이테크, 휴니드테크놀러지스, 대원전선, 선도전기, 대동전자, 아남전자, 경인전자, 삼화전기, 삼화전자공업, 와이투솔루션, 성문전자, 에이엔피, 광전자, 티에이치엔, 엑시큐어하이트론, 일진디스플, 케이엠더블유, 시그네틱스, 주연테크, 리노공업, 에스피지, 신성델타테크, 다원시스, 이엠텍, 파트론, KEC


"C, D등급 대상 컨설팅 시 어떤 기업을 모범사례(Best Practice)로 제시해야 하는가?"


삼성전자, SK하이닉스 같은 최상위 A+ 대기업보다 C/D에서 B+~A로 등급이 급상승한 중견/중소 턴어라운드(Turnaround) 기업을 벤치마킹 대상으로 잡아야 합니다.

최상위 기업(A+)의 한계: 초대형 대기업은 전담 조직과 막대한 예산을 투입하므로, C/D등급 기업이 보기엔 "현실적으로 따라 할 수 없는 먼 나라 이야기"로 느껴집니다.



---



###턴어라운드 기업 모범사례(추천):

보고서 미발행 → 첫 발행 후 B+ 이상 직행한 C26 업종 내 기업

환경(E)이나 지배구조(G) 세부 항목에서 특정 연도에 등급이 2단계 이상 뛰어오른 기업

컨설팅 전략: "이 기업도 2년 전엔 C등급이었고 보고서도 없었지만, [보고서 첫 공시 + 특정 세부지표 개선]을 통해 1년 만에 B+로 진입했다"는 현실적 로드맵(Gap Analysis)을 제시할 때 고객사의 설득력이 가장 높습니다.

In [ ]:
import pandas as pd

# 1. 미발행 49개사 목록 정의
companies_49 = [
    '에스앤에스텍', 'LX세미콘', '주성코퍼레이션', '하나머티리얼즈', '해성디에스',
    '제이앤티씨', '덕산네오룩스', 'RFHIC', '심텍', '솔루엠', '한화시스템',
    '이녹스첨단소재', '솔루스첨단소재', '대덕전자', '더블유씨피', '가온칩스',
    '에코프로머티', 'SK하이닉스', 'DB하이텍', '금호전기', '삼화콘덴서공업',
    '써니전자', '씨아이테크', '휴니드테크놀러지스', '대원전선', '선도전기',
    '대동전자', '아남전자', '경인전자', '삼화전기', '삼화전자공업', '와이투솔루션',
    '성문전자', '에이엔피', '광전자', '티에이치엔', '엑시큐어하이트론', '일진디스플',
    '케이엠더블유', '시그네틱스', '네패스', 'SFA반도체', '에프에스티', '주연테크',
    '서울반도체', '인탑스', '인터플렉스', '리노공업', '에스피지'
]

# 2. 2025년 평가 데이터 중 49개사 추출
df_49 = df[(df['평가년도'] == 2025) & (df['기업명'].isin(companies_49))][['기업명', 'ESG등급', '시장구분']].drop_duplicates(subset=['기업명'])

# 3. 등급 정렬 순서 정의 (A -> B+ -> B -> C -> D)
grade_order = ['A', 'B+', 'B', 'C', 'D']
df_49['ESG등급'] = pd.Categorical(df_49['ESG등급'], categories=grade_order, ordered=True)

# ---------------------------------------------------------
# [출력 1] ESG 등급 x 시장구분 교차 집계표 (피벗 테이블)
# ---------------------------------------------------------
ct = pd.crosstab(df_49['ESG등급'], df_49['시장구분'], margins=True, margins_name='합계')
print("=" * 60)
print("■ 1. ESG 등급 x 시장구분 교차 수량 집계 (49개사)")
print("=" * 60)
print(ct)
print("\n")

# ---------------------------------------------------------
# [출력 2] 등급별 & 시장구분별 상세 기업 목록
# ---------------------------------------------------------
print("=" * 60)
print("■ 2. 등급별 KOSPI / KOSDAQ 기업 상세 목록")
print("=" * 60)

grouped = df_49.groupby('ESG등급', observed=False)

for grade, group in grouped:
    kospi_list = group[group['시장구분'] == 'KOSPI']['기업명'].tolist()
    kosdaq_list = group[group['시장구분'] == 'KOSDAQ']['기업명'].tolist()

    total_cnt = len(group)
    print(f"\n▶ [{grade} 등급] - 총 {total_cnt}개사")
    print(f"   - KOSPI  ({len(kospi_list)}개사): {', '.join(kospi_list) if kospi_list else '없음'}")
    print(f"   - KOSDAQ ({len(kosdaq_list)}개사): {', '.join(kosdaq_list) if kosdaq_list else '없음'}")

■ 1. ESG 등급 x 시장구분 교차 수량 집계 (49개사)
시장구분   KOSDAQ  KOSPI  합계
ESG등급                   
A           1      4   5
B+          2      4   6
B           1      0   1
C           7      2   9
D           8     20  28
합계         19     30  49


■ 2. 등급별 KOSPI / KOSDAQ 기업 상세 목록

▶ [A 등급] - 총 5개사
   - KOSPI  (4개사): LX세미콘, 해성디에스, 한화시스템, SK하이닉스
   - KOSDAQ (1개사): 심텍

▶ [B+ 등급] - 총 6개사
   - KOSPI  (4개사): 솔루스첨단소재, 대덕전자, 에코프로머티, DB하이텍
   - KOSDAQ (2개사): 이녹스첨단소재, 인탑스

▶ [B 등급] - 총 1개사
   - KOSPI  (0개사): 없음
   - KOSDAQ (1개사): 하나머티리얼즈

▶ [C 등급] - 총 9개사
   - KOSPI  (2개사): 솔루엠, 삼화콘덴서공업
   - KOSDAQ (7개사): 에스앤에스텍, 덕산네오룩스, 네패스, SFA반도체, 에프에스티, 서울반도체, 인터플렉스

▶ [D 등급] - 총 28개사
   - KOSPI  (20개사): 주성코퍼레이션, 금호전기, 써니전자, 씨아이테크, 휴니드테크놀러지스, 대원전선, 선도전기, 대동전자, 아남전자, 경인전자, 삼화전기, 삼화전자공업, 와이투솔루션, 성문전자, 에이엔피, 광전자, 티에이치엔, 엑시큐어하이트론, 일진디스플, 주연테크
   - KOSDAQ (8개사): 제이앤티씨, RFHIC, 더블유씨피, 가온칩스, 케이엠더블유, 시그네틱스, 리노공업, 에스피지


In [ ]:
import pandas as pd

# ----------------------------------------------------
# 1. 대상 기업(C26 업종 76개사) 필터링
# ----------------------------------------------------
c26_companies = [
    '에스앤에스텍', 'LX세미콘', '주성코퍼레이션', '하나머티리얼즈', '서진시스템', '유티아이', '인텔리안테크', '드림텍',
    '해성디에스', '제이앤티씨', '덕산네오룩스', 'RFHIC', '심텍', '솔루엠', '한화시스템', '이녹스첨단소재',
    '솔루스첨단소재', '대덕전자', '더블유씨피', '가온칩스', '에코프로머티', 'SK하이닉스', 'DB하이텍', '금호전기',
    '삼화콘덴서공업', '세방전지', '한솔테크닉스', '써니전자', '씨아이테크', '삼영전자공업', '휴니드테크놀러지스', '삼성전자',
    '대원전선', '선도전기', '이수페타시스', '코리아써키트', '대동전자', '아남전자', '경인전자', '삼성전기',
    '경동나비엔', '삼화전기', '엘에스일렉트릭', 'LG이노텍', '삼화전자공업', '와이투솔루션', '신성이엔지', '성문전자',
    '에이엔피', '광전자', '티에이치엔', '엑시큐어하이트론', '롯데에너지머티리얼즈', '일진디스플', '시노펙스', '한국단자공업',
    '신도리코', '케이엠더블유', '시그네틱스', '자화전자', '네패스', 'LG디스플레이', 'SFA반도체', '에프에스티',
    '주연테크', '서울반도체', '인탑스', '인터플렉스', '리노공업', '에스피지', '신성델타테크', '하나마이크론',
    '다원시스', '대주전자재료', '제주반도체', '아이티엠반도체', '비에이치', '이엠텍', '파트론', 'KEC', '칩스앤미디어', '엠씨넥스'
]

df_c26 = df[(df['기업명'].isin(c26_companies)) & (df['평가년도'].between(2022, 2025))].copy()

# 2025년 신규 편입 6개사 제외 모수 76개사
pv_check = pd.pivot_table(df_c26, index='기업명', columns='평가년도', values='ESG등급', aggfunc='count').fillna(0)
only_2025 = pv_check[(pv_check[2022]==0) & (pv_check[2023]==0) & (pv_check[2024]==0) & (pv_check[2025]==1)].index.tolist()
target_76 = [c for c in c26_companies if c not in only_2025]

df_target = df_c26[df_c26['기업명'].isin(target_76)].copy()

# ----------------------------------------------------
# 2. 등급 수치화 매핑 (변화량 계산용)
# ----------------------------------------------------
grade_map = {'S': 7, 'A+': 6, 'A': 5, 'B+': 4, 'B': 3, 'C': 2, 'D': 1}
cols_to_map = ['ESG등급', '환경', '사회', '지배구조']

for col in cols_to_map:
    if col in df_target.columns:
        df_target[f'{col}_num'] = df_target[col].map(grade_map).fillna(0)

# ----------------------------------------------------
# 3. 턴어라운드 유형별 감지 함수
# ----------------------------------------------------

# (1) 보고서 공시 전환 턴어라운드 (N -> Y 전환)
pv_sus = df_target.pivot_table(index='기업명', columns='평가년도', values='지속가능경영보고서_발행', aggfunc='first').fillna('N')
pv_cg = df_target.pivot_table(index='기업명', columns='평가년도', values='기업지배구조보고서_발행', aggfunc='first').fillna('N')

# 과거(2022~2024)엔 둘 다 N이었다가 2025년에 하나 이상 Y로 바뀐 기업
first_report_comps = []
for comp in target_76:
    past_sus = (pv_sus.loc[comp, [2022, 2023, 2024]] == 'Y').any() if comp in pv_sus.index else False
    past_cg = (pv_cg.loc[comp, [2022, 2023, 2024]] == 'Y').any() if comp in pv_cg.index else False

    curr_sus = pv_sus.loc[comp, 2025] == 'Y' if comp in pv_sus.index else False
    curr_cg = pv_cg.loc[comp, 2025] == 'Y' if comp in pv_cg.index else False

    # 보고서를 공시하기 시작한 기업
    if not (past_sus or past_cg) and (curr_sus or curr_cg):
        first_report_comps.append(comp)

# (2) 종합 ESG 등급 2단계 이상 상승 기업 (예: C -> B+ 이상)
pv_esg_num = df_target.pivot_table(index='기업명', columns='평가년도', values='ESG등급_num', aggfunc='first').fillna(0)
pv_esg_str = df_target.pivot_table(index='기업명', columns='평가년도', values='ESG등급', aggfunc='first').fillna('-')

turnaround_esg = []
for comp in pv_esg_num.index:
    start_score = pv_esg_num.loc[comp, 2022] if 2022 in pv_esg_num.columns else 0
    end_score = pv_esg_num.loc[comp, 2025] if 2025 in pv_esg_num.columns else 0

    # 최초 등급 존재 + 2단계 이상 상승
    if start_score > 0 and (end_score - start_score) >= 2:
        diff = int(end_score - start_score)
        turnaround_esg.append({
            '기업명': comp,
            '2022년': pv_esg_str.loc[comp, 2022],
            '2025년': pv_esg_str.loc[comp, 2025],
            '상승단계': f"+{diff}단계"
        })

df_turnaround_esg = pd.DataFrame(turnaround_esg)

# ----------------------------------------------------
# 4. 결과 분석 출력
# ----------------------------------------------------
print("=" * 75)
print("■ C26 업종 컨설팅용 턴어라운드(Turnaround) 모범사례 기업 추출 결과")
print("=" * 75)

print(f"\n1. [보고서 공시 전환 턴어라운드] (2022~2024 미발행 -> 2025년 첫 공시)")
print(f"   • 대상 기업 ({len(first_report_comps)}개사): {', '.join(first_report_comps) if first_report_comps else '해당 없음'}")
if first_report_comps:
    print("   • 추천 활용: '보고서 공시 시작만으로 등급 상승 기틀을 마련한 대표 사례'")

print(f"\n2. [ESG 종합등급 급상승 턴어라운드] (2022년 대비 2025년 +2단계 이상 상승)")
if not df_turnaround_esg.empty:
    print(df_turnaround_esg.to_string(index=False))
    print("   • 추천 활용: 'C/D 등급에서 2~3년 만에 B+/A 등급으로 진입한 실질 개선 벤치마킹 사례'")
else:
    print("   • 2단계 이상 상승한 기업이 없습니다.")

print("\n" + "=" * 75)

■ C26 업종 컨설팅용 턴어라운드(Turnaround) 모범사례 기업 추출 결과

1. [보고서 공시 전환 턴어라운드] (2022~2024 미발행 -> 2025년 첫 공시)
   • 대상 기업 (0개사): 해당 없음

2. [ESG 종합등급 급상승 턴어라운드] (2022년 대비 2025년 +2단계 이상 상승)
    기업명 2022년 2025년 상승단계
  경동나비엔     C    B+ +2단계
 대주전자재료     D    B+ +3단계
    드림텍     D     B +2단계
솔루스첨단소재     C    B+ +2단계
     심텍     C     A +3단계
이녹스첨단소재     C    B+ +2단계
하나머티리얼즈     D     B +2단계
 한국단자공업     D     B +2단계
   • 추천 활용: 'C/D 등급에서 2~3년 만에 B+/A 등급으로 진입한 실질 개선 벤치마킹 사례'



분석은 하되 '최상위 가이드라인(Gap 측정 기준)' 용도로 활용해야 합니다.

우선 분석의 목적: A/A+ 기업 분석은 "이들처럼 되라"고 보여주는 목적이 아니라, ESG 평가항목(E/S/G 각 지표)에서 감점을 받지 않기 위해 갖춰야 할 '필수 요건(Baseline Checklist)'이 무엇인지 파악하기 위해 필요합니다.



---

추천 분석 방식:

A/A+ 기업들이 보고서에서 공통적으로 공시하고 있는 핵심 지표(온실가스 Scope 1,2,3, 이사회 독립성, 안전보건 인증 등)를 추출합니다.

이를 C/D등급 기업의 공시 수준과 비교하여 '필수 공시 항목 Gap'을 데이터 테이블로 시각화해 줍니다.

###격차분석 및 보고서 발행 효과 측정

In [ ]:
import pandas as pd
import plotly.graph_objects as go
from plotly.subplots import make_subplots

# ----------------------------------------------------
# 1. 대상 데이터 필터링 및 수치화 매핑
# ----------------------------------------------------
c26_companies = [
    '에스앤에스텍', 'LX세미콘', '주성코퍼레이션', '하나머티리얼즈', '서진시스템', '유티아이', '인텔리안테크', '드림텍',
    '해성디에스', '제이앤티씨', '덕산네오룩스', 'RFHIC', '심텍', '솔루엠', '한화시스템', '이녹스첨단소재',
    '솔루스첨단소재', '대덕전자', '더블유씨피', '가온칩스', '에코프로머티', 'SK하이닉스', 'DB하이텍', '금호전기',
    '삼화콘덴서공업', '세방전지', '한솔테크닉스', '써니전자', '씨아이테크', '삼영전자공업', '휴니드테크놀러지스', '삼성전자',
    '대원전선', '선도전기', '이수페타시스', '코리아써키트', '대동전자', '아남전자', '경인전자', '삼성전기',
    '경동나비엔', '삼화전기', '엘에스일렉트릭', 'LG이노텍', '삼화전자공업', '와이투솔루션', '신성이엔지', '성문전자',
    '에이엔피', '광전자', '티에이치엔', '엑시큐어하이트론', '롯데에너지머티리얼즈', '일진디스플', '시노펙스', '한국단자공업',
    '신도리코', '케이엠더블유', '시그네틱스', '자화전자', '네패스', 'LG디스플레이', 'SFA반도체', '에프에스티',
    '주연테크', '서울반도체', '인탑스', '인터플렉스', '리노공업', '에스피지', '신성델타테크', '하나마이크론',
    '다원시스', '대주전자재료', '제주반도체', '아이티엠반도체', '비에이치', '이엠텍', '파트론', 'KEC', '칩스앤미디어', '엠씨넥스'
]

# 2025년 기준 데이터 필터링
df_2025 = df[(df['기업명'].isin(c26_companies)) & (df['평가년도'] == 2025)].copy()

# 등급 수치 변환 (D=1 ~ S=7)
grade_map = {'S': 7, 'A+': 6, 'A': 5, 'B+': 4, 'B': 3, 'C': 2, 'D': 1}
cols = ['ESG등급', '환경', '사회', '지배구조']
for c in cols:
    df_2025[f'{c}_num'] = df_2025[c].map(grade_map).fillna(1)

# 그룹 분류
df_2025['그룹'] = 'B~B+ 등급'
df_2025.loc[df_2025['ESG등급'].isin(['A+', 'A']), '그룹'] = 'A/A+ 우수군'
df_2025.loc[df_2025['ESG등급'].isin(['C', 'D']), '그룹'] = 'C/D 취약군'

df_2025['보고서발행'] = '미발행(N)'
df_2025.loc[(df_2025['지속가능경영보고서_발행'] == 'Y') | (df_2025['기업지배구조보고서_발행'] == 'Y'), '보고서발행'] = '1개 이상 발행(Y)'

# ----------------------------------------------------
# 2. 레이더 차트 (A/A+ vs C/D 영역별 격차)
# ----------------------------------------------------
group_avg = df_2025.groupby('그룹')[['환경_num', '사회_num', '지배구조_num']].mean()

categories = ['환경 (E)', '사회 (S)', '지배구조 (G)']

fig1 = go.Figure()

# A/A+ 우수군
fig1.add_trace(go.Scatterpolar(
    r=group_avg.loc['A/A+ 우수군'].values.tolist() + [group_avg.loc['A/A+ 우수군'].values[0]],
    theta=categories + [categories[0]],
    fill='toself',
    name='A/A+ 우수군',
    line_color='#1F4E79'
))

# C/D 취약군
fig1.add_trace(go.Scatterpolar(
    r=group_avg.loc['C/D 취약군'].values.tolist() + [group_avg.loc['C/D 취약군'].values[0]],
    theta=categories + [categories[0]],
    fill='toself',
    name='C/D 취약군',
    line_color='#D9534F'
))

fig1.update_layout(
    polar=dict(radialaxis=dict(visible=True, range=[1, 7], tickvals=[1,2,3,4,5,6,7], ticktext=['D','C','B','B+','A','A+','S'])),
    title="<b>[관점 1] A/A+ 우수군 vs C/D 취약군 E/S/G 영역별 평균 등급 격차</b>",
    font=dict(family="NanumGothic, Apple SD Gothic Neo, Malgun Gothic", size=12),
    width=700, height=500
)

try:
    fig1.show(renderer="colab")
except:
    fig1.show()

# ----------------------------------------------------
# 3. 바 차트 (보고서 발행 여부에 따른 E/S/G 점수 격차)
# ----------------------------------------------------
rep_avg = df_2025.groupby('보고서발행')[['ESG등급_num', '환경_num', '사회_num', '지배구조_num']].mean().reset_index()

fig2 = go.Figure()
categories_bar = ['종합 ESG', '환경 (E)', '사회 (S)', '지배구조 (G)']

fig2.add_trace(go.Bar(
    x=categories_bar,
    y=rep_avg[rep_avg['보고서발행'] == '1개 이상 발행(Y)'][['ESG등급_num', '환경_num', '사회_num', '지배구조_num']].values[0],
    name='보고서 발행(Y)',
    marker_color='#2F75B5',
    text=[f"{v:.1f}점" for v in rep_avg[rep_avg['보고서발행'] == '1개 이상 발행(Y)'][['ESG등급_num', '환경_num', '사회_num', '지배구조_num']].values[0]],
    textposition='auto'
))

fig2.add_trace(go.Bar(
    x=categories_bar,
    y=rep_avg[rep_avg['보고서발행'] == '미발행(N)'][['ESG등급_num', '환경_num', '사회_num', '지배구조_num']].values[0],
    name='보고서 미발행(N)',
    marker_color='#C55A11',
    text=[f"{v:.1f}점" for v in rep_avg[rep_avg['보고서발행'] == '미발행(N)'][['ESG등급_num', '환경_num', '사회_num', '지배구조_num']].values[0]],
    textposition='auto'
))

fig2.update_layout(
    barmode='group',
    title="<b>[관점 2] 보고서 발행 여부(Y/N)에 따른 영역별 등급 수치 격차</b><br><sup>* 점수 기준: D(1점) ~ A+(6점) / S(7점)</sup>",
    yaxis=dict(title="평균 등급 수치", range=[0, 7]),
    font=dict(family="NanumGothic, Apple SD Gothic Neo, Malgun Gothic", size=12),
    width=800, height=480
)

try:
    fig2.show(renderer="colab")
except:
    fig2.show()

###인사이트

### 1. [관점 1] C/D 취약군의 최대 아킬레스건은 ‘지배구조(G)’와 ‘환경(E)’

* **데이터 인사이트**: 레이더 차트에서 C/D 취약군은 사회(S) 영역에 비해 **환경(E)과 지배구조(G) 영역의 면적이 현저히 축소**되어 나타납니다.
* **컨설팅 시사점**:
* 중소·중견기업 다수가 노동/안전/사회공헌 등 **사회(S) 영역**에서는 기본적 법규 준수로 일정 점수를 받지만, **환경(E, 온실가스·폐기물 데이터)** 및 **지배구조(G, 이사회 투명성·주주권익)** 영역에서는 공시 및 제도 부재로 대거 감점을 당하고 있습니다.
* 따라서 C/D 기업 대상 컨설팅 시 "E와 G 항목의 공시 체계 구축"을 최우선 개선 과제(Top Priority)로 지정해야 합니다.



---

### 2. [관점 2] 보고서 발행(Y)은 '평가 등급을 한 단계 격상'시키는 가장 확실한 레버리지

* **데이터 인사이트**: 보고서를 1개 이상 발행한 기업군(Y)의 종합 ESG 및 영역별 평균 점수는 미발행 기업군(N) 대비 **약 1.5 ~ 2.0점(등급 기준 C/D → B/B+ 이상) 이상 압도적으로 높습니다.**
* **컨설팅 시사점**:
* C/D 기업의 낮은 등급 원인은 실제 ESG 경영 수행 여부보다 "공시 부재로 인한 평가 제외(Information Gap)"에 더 큽니다.
* 경영진을 설득할 때 "보고서 발간 하나만으로도 즉시 C/D 탈출 및 B+ 진입이 가능하다"는 명확한 ROI(투자 대비 효과) 근거로 제시할 수 있습니다.



---

### 3. 영역별 진입 장벽 구조 (G > E > S)

* **데이터 인사이트**: 미발행 기업(N)이 발행 기업(Y)으로 전환할 때 가장 격차가 크게 벌어지는 지표는 지배구조(G)이며, 그다음이 환경(E)입니다.
* **컨설팅 시사점**:
* **지배구조(G)**: '기업지배구조보고서'의 핵심 지표(사외이사 비율, 세부 위원회 설치, 주총 소집 통지 등) 표준화 및 공시 절차 수립이 시급합니다.
* **환경(E)**: Scope 1, 2 온실가스 배출량 산정 및 에너지 정량 데이터 정비 체계(지속가능경영보고서 인덱스 대응)를 조기 확보해야 합니다.



---

### 💡 컨설팅 제안서 한 줄 요약 (Executive Summary)

> **"C26 업종 C/D등급 기업의 핵심 문제는 '경영 성과'가 아닌 '공시 부재(Information Gap)'에 있으며, 보고서 제정 및 E·G 영역 중심의 정량 공시 체계 구축만으로도 B+ 이상 턴어라운드가 가능합니다."**

"챗봇을 구현해서 질의응답 시스템을 만드는 것이 목표가 되어야 하는가?"
단순 챗봇 구현 자체를 최종 목표로 잡기보다는 ESG 진단 & 로드맵 생성형 AI 솔루션 형태로 연계하는 것을 강력히 권장합니다.

단순 챗봇의 한계: "GRI 가이드라인이 뭐야?" 같은 일반 질의응답 챗봇은 기존 LLM(ChatGPT 등)으로도 대체 가능하므로 컨설팅 데이터 프로젝트로서의 차별성이 떨어집니다.



---



컨설팅 프로젝트에 적합한 AI 활용안 (RAG 기반 ESG Advisor):

[입력]: C/D등급 기업의 현재 ESG 데이터(또는 보고서 미발행 여부) 입력.

[AI 로직]: 지금까지 분석하신 C26 업종 76개사 ESG 등급 DB 및 A/A+ 기업 벤치마킹 데이터를 바탕으로 RAG(검색증강생성) 수행.

[출력]:

"귀사는 현재 보고서 미발행으로 인해 최소 X등급 손실을 보고 있습니다."

"우선순위 개선 항목 TOP 3: ① 지속가능경영보고서 발간, ② 환경경영시스템 인증, ③ 이사회 투명성 강화"

"유사 턴어라운드 모범사례: OO기업의 B+ 진입 전략 참조"